### PR TITLE
Feature/add calendar visualisation

### PR DESCRIPTION
**Pull request by irfankhan10**
# Description

Added a function to visualise calendar data using matplotlib + update to the queue visibility article

# Checklist

- [x] I have performed a self-review of my code.
- [x] I have commented hard-to-understand parts of my code.
- [x] I have made corresponding changes to the public API documentation.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have updated the changelog with any user-facing changes.

### CODE REVIEW

File: pytket/extensions/quantinuum/backends/calendar_visualisation.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
+"""Submodule providing calendar visualisation functionality"""
```
Comment (cqc-alec) [inline pytket/extensions/quantinuum/backends/calendar_visualisation.py:1]: Please add copyright notice.

File: pytket/extensions/quantinuum/backends/quantinuum.py [hunk 2]
```diff
@@ -452,14 +462,14 @@ def get_calendar(
         start_date: datetime.datetime,
         end_date: datetime.datetime,
         localise: bool = True,
-    ) -> List[Dict[str, str]]:
-        r"""Retrieves the Quantinuum H-Series operational calendar
+    ) -> List[Dict[str, object]]:
```
Comment (cqc-alec) [inline pytket/extensions/quantinuum/backends/quantinuum.py:465]: `Any` instead of `object`?

File: setup.py [hunk 3]
```diff
@@ -54,6 +54,7 @@
     ],
     extras_require={
         "pecos": ["pytket-pecos ~= 0.1.18"],
+        "calendar": ["matplotlib ~= 3.8.3", "pandas ~= 2.2.1"],
```
Comment (cqc-alec) [inline setup.py:57]: Documentation about this functionality and how to enable it should be added to the front documentation page.

**Review by cqc-alec — CHANGES_REQUESTED** [review]
Looks like there are some merge conflicts to resolve.

File: setup.py (hunk 3 above)
Comment (irfankhan10) [reply setup.py:57]: Added to the front page as requested

File: tests/integration/backend_test.py [hunk 4]
```diff
@@ -1399,3 +1403,23 @@ def test_get_calendar_raises_error(
...
+
+
+@pytest.mark.skipif(skip_mpl_tests, reason=REASON_MPL)
+@pytest.mark.timeout(120)
+@pytest.mark.mpl_image_compare
+def test_view_calendar(authenticated_quum_handler: QuantinuumAPI) -> Any:
+    backend = QuantinuumBackend(
+        api_handler=authenticated_quum_handler, device_name="H1-1"
+    )
+    return backend.view_calendar(month=2, year=2024)
```
Comment (cqc-alec) [inline tests/integration/backend_test.py:1425]: Do you know why this test is failing?

File: pytket/extensions/quantinuum/backends/calendar_visualisation.py (hunk 1 above)
Comment (cqc-alec) [reply pytket/extensions/quantinuum/backends/calendar_visualisation.py:18]: Let's wait for confirmation that this is the right format/

File: tests/integration/backend_test.py (hunk 4 above)
Comment (irfankhan10) [reply tests/integration/backend_test.py:1425]: yes, the data returned does not match the data in the png file. We need to use normal credentials. In other words, it should not be a  QA test

File: docs/intro.txt [hunk 5]
```diff
@@ -15,6 +15,15 @@ and Windows. To install, run:
             The user will be prompted for their login credentials when making API calls such as calling the ``process_circuits`` method or querying ``backend_info``.
 
 
+``pytket-quantinuum`` provides calendar visualization capabilities. The ``calendar`` extra-install-argument must be specified to install matplotlib and pandas.
+
+::
+
+    pip install pytket-quantinuum["calendar"]
```
Comment (cqc-alec) [inline docs/intro.txt:22]: I don't think we want quotes here?

Comment (irfankhan10) [reply docs/intro.txt:22]: should be around pytket-quantinuum. I will change

Comment (cqc-alec) [reply docs/intro.txt:22]: > should be around pytket-quantinuum. I will change

Not needed around pytket-quantinuum either.

Comment (irfankhan10) [reply docs/intro.txt:22]: OK. In the past, I ran into issues not having the quotes, but maybe this is because of zsh shell.

**Review by cqc-alec — APPROVED** [review]

**Pull request merged**